src/crimson/osd/osd.cc

```diff
@@ -669,6 +669,9 @@ OSD::ms_dispatch(crimson::net::ConnectionRef conn, MessageRef m)
   if (pg_shard_manager.is_stopping()) {
     return {};
   }
+  // XXX: we're assuming the `switch` part is executed immediately, and
```

Comment:
will we be able to notice the problem immediately, if that fact changes in some future?

Reply:
Nope :-(. We should rework.